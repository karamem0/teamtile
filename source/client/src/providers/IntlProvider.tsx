//
// Copyright (c) 2021-2024 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/teamtile/blob/main/LICENSE
//

import React from 'react';

import {
  IntlProvider as Provider,
  createIntl,
  MessageFormatElement
} from 'react-intl';

import en from '../translations/compiled/en.json';
import ja from '../translations/compiled/ja.json';

const translations: { [key: string]: Record<string, MessageFormatElement[]> } = {
  en,
  ja
};

const intl = createIntl({
  defaultLocale: 'en',
  locale: window.navigator.language.substring(0, 2),
  messages: translations[window.navigator.language.substring(0, 2)]
});

function IntlProvider(props: React.PropsWithChildren<unknown>) {

  const { children } = props;

  return (
    <Provider {...intl}>
      {children}
    </Provider>
  );

}

export default IntlProvider;
