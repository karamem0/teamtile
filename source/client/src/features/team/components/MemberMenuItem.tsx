//
// Copyright (c) 2021-2025 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/teamtile/blob/main/LICENSE
//

import React from 'react';

import { useDrawer } from '../../../common/providers/DrawerProvider';

import Presenter from './MemberMenuItem.presenter';

interface MemberMenuItemProps {
  id?: string
}

function MemberMenuItem(props: Readonly<MemberMenuItemProps>) {

  const { id } = props;

  const { setDrawer } = useDrawer();

  const handleClick = React.useCallback(() => {
    setDrawer({
      data: id,
      type: 'member'
    });
  }, [
    id,
    setDrawer
  ]);

  return (
    <Presenter onClick={handleClick} />
  );

}

export default MemberMenuItem;
