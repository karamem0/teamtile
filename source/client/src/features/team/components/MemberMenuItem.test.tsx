//
// Copyright (c) 2023 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/teamtile/blob/main/LICENSE
//

import React from 'react';

import { render, screen } from '@testing-library/react';

import IntlProvider from '../../../providers/IntlProvider';

import MemberMenuItem from './MemberMenuItem.presenter';

test('create shapshot', async () => {
  const params = {
    items: [
      {
        id: '87d349ed-44d7-43e1-9a83-5f2406dee5bd',
        displayName: 'Adele Vance',
        userId: '87d349ed-44d7-43e1-9a83-5f2406dee5bd',
        email: 'AdeleV@M365x214355.onmicrosoft.com'
      }
    ]
  };
  render(
    <IntlProvider>
      <MemberMenuItem {...params} />
    </IntlProvider>
  );
  expect(screen.queryAllByText(/^.*$/)[0]).toMatchSnapshot();
});
