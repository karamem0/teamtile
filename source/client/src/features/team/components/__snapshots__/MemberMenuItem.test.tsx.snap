// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should match the snapshot 1`] = `
<DocumentFragment>
  .emotion-0 {
  min-height: 100vh;
  line-height: 1.25em;
  background-color: #f5f5f5;
}

<div
    class="fui-FluentProvider fui-FluentProviderr0 ___jdtuxv0_17k0bs4 f19n0e5 fxugw4r f1o700av fk6fouc fkhj508 figsok6 f1i3iumi"
    dir="ltr"
  >
    <div
      class="emotion-0"
    >
      <div
        class="fui-MenuItem rfoezjv"
        role="menuitem"
        tabindex="0"
      >
        <span
          class="fui-MenuItem__icon ro9koqv"
        >
          <svg
            aria-hidden="true"
            class="___12fm75w_1fmvgpr f1w7gpdv fez10in fg4l7m0"
            fill="currentColor"
            height="16"
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M7.5 9a2 2 0 0 1 2 2c0 .97-.6 1.73-1.41 2.23-.82.5-1.91.77-3.09.77a6.02 6.02 0 0 1-3.09-.77C1.1 12.73.5 11.97.5 11c0-1.1.9-2 2-2h5Zm-5 1a1 1 0 0 0-1 1c0 .5.3 1 .93 1.38.63.38 1.54.62 2.57.62 1.03 0 1.94-.24 2.57-.63.63-.38.93-.86.93-1.37a1 1 0 0 0-1-1h-5Zm11.65-1a1.5 1.5 0 0 1 1.35 1.5c0 .77-.47 1.4-1.1 1.83-.64.42-1.49.67-2.4.67a4.6 4.6 0 0 1-1.92-.4c.16-.29.28-.6.35-.95.43.21.97.35 1.57.35.74 0 1.4-.2 1.84-.5.46-.3.66-.67.66-1a.5.5 0 0 0-.4-.49L14 10h-3.67a3 3 0 0 0-.58-.98L10 9h4.15Zm-1.92-5.5a2.25 2.25 0 0 1 2.02 2.25v.23A2.25 2.25 0 0 1 12 8L11.77 8a2.25 2.25 0 0 1-2-2.01l-.02-.23c0-1.24 1-2.25 2.25-2.25h.23ZM5 2.5A2.75 2.75 0 1 1 5 8a2.75 2.75 0 0 1 0-5.5Zm7 2A1.25 1.25 0 1 0 12 7a1.25 1.25 0 0 0 0-2.5Zm-7-1A1.75 1.75 0 1 0 5 7a1.75 1.75 0 0 0 0-3.5Z"
              fill="currentColor"
            />
          </svg>
        </span>
        <span
          class="fui-MenuItem__content r1ls86vo"
        >
          View members
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
